src/Common/src/Interop/Unix/System.Native/Interop.FdSet.cs

```diff
@@ -2,34 +2,54 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
+using System.Diagnostics;
 using System.Runtime.InteropServices;
 
 internal static partial class Interop
 {
     internal static partial class Sys
     {
-        public unsafe struct FdSet
-        {
-            const int FDSET_MAX_FDS = 1024;
-            const int FDSET_NFD_BITS = FDSET_MAX_FDS / (8 * sizeof(uint));
+        [DllImport(Libraries.SystemNative)]
+        private static extern int FdSetSize();
+
+        private static readonly int FD_SETSIZE_BITS = FdSetSize();
+
+        private const int BitsPerByte = 8;
+        private static readonly int FD_SETSIZE_BYTES = FD_SETSIZE_BITS / BitsPerByte;
+
+        internal static readonly int FD_SETSIZE_UINTS = FD_SETSIZE_BYTES / sizeof(uint);
 
-            private fixed uint _bits[FDSET_MAX_FDS / FDSET_NFD_BITS];
-  
-            public void Set(int fd)
+        internal static unsafe void FD_SET(int fd, uint* fdset)
+        {
+            if (fd >= FD_SETSIZE_BITS)
```

Comment:
>  I'm not sure that this is a complete solution

What problem do you believe still remains?  Is your concern that there could legitimately be an fd value that's >= FD_SETSIZE?

>  I wonder if we should switch to using poll under the covers, which allows arbitrary file descriptor numbers at the expense of a larger memory footprint

If we don't need to stick with select, I'm fine with that, and can close this in you want to switch over to poll (or maybe merge this and then switch over to poll).